clang/lib/Sema/HLSLBuiltinTypeDeclBuilder.h

```diff
@@ -52,6 +52,7 @@ class BuiltinTypeDeclBuilder {
   ClassTemplateDecl *PrevTemplate = nullptr;
   NamespaceDecl *HLSLNamespace = nullptr;
   llvm::StringMap<FieldDecl *> Fields;
+  bool IsConstResourceClass = false;
```

Comment:
We don't seem to keep the resource class around in the Builder, so I added this value to track it.

If there is a better way to derive the value/access the resource class please lmk!

Reply:
Let's go with storing the RC, that seems sensible to me.